blivet/populator/helpers/disk.py

```diff
@@ -214,3 +214,28 @@ def _get_kwargs(self):
 
         log.info("%s is a zfcp device", udev.device_get_name(self.data))
         return kwargs
+
+
+class NVDIMMNamespaceDevicePopulator(DiskDevicePopulator):
+    priority = 20
+
+    _device_class = NVDIMMNamespaceDevice
+
+    @classmethod
+    def match(cls, data):
+        return (super(NVDIMMNamespaceDevicePopulator, NVDIMMNamespaceDevicePopulator).match(data) and
+                udev.device_is_nvdimm_namespace(data))
+
+    def _get_kwargs(self):
+        kwargs = super(NVDIMMNamespaceDevicePopulator, self)._get_kwargs()
+
+        from ...nvdimm import nvdimm
+        ninfo = nvdimm.get_namespace_info(self.data.get("DEVNAME"))
+
+        kwargs["mode"] = blockdev.nvdimm_namespace_get_mode_str(ninfo.mode)
+        kwargs["devname"] = ninfo.dev
+        kwargs["uuid"] = ninfo.uuid
+        kwargs["sector_size"] = ninfo.sector_size
```

Comment:
Is the sector size used for anything?

Reply:
No, it isn't. I thought it might be useful just as a "property", because it is possible to change/set sector size when creating the namespace.